src/Microsoft.PowerShell.Commands.Diagnostics/GetEventCommand.cs

```diff
@@ -1538,6 +1542,41 @@ private string HandleDataHashValue(Object value)
 
 
         //
+        // HandleNamedDataHashValue helper for hashtable structured query builder.
+        // Constructs and returns named event data field XPath portion as a string.
+        // Fix Issue #2327
+        //
+        private string HandleNamedDataHashValue(String key, Object value)
+        {
+            string ret = "";
+            if (value is Array)
+            {
+                Array dataArray = (Array)(value);
+                ret += "(";
+                for (int i = 0; i < dataArray.Length; i++)
+                {
+                    ret += string.Format(CultureInfo.InvariantCulture,
+                                         "((EventData[Data[@Name='{0}']='{1}']) or (UserData/*/{0}='{1}'))",
+                                         key, dataArray.GetValue(i).ToString());
+                    if (i < (dataArray.Length - 1))
+                    {
+                        ret += " or ";
+                    }
+                }
+                ret += ")";
+            }
+            else
+            {
+                ret += string.Format(CultureInfo.InvariantCulture,
+                                         "((EventData[Data[@Name='{0}']='{1}']) or (UserData/*/{0}='{1}'))",
```

Comment:
> "((EventData[Data[@Name='{0}']='{1}']) or (UserData/*/{0}='{1}'))"

Let's not repeat this twice. Maybe it's better to hold this format in a const variable.